emscripten.py

```diff
@@ -709,10 +709,13 @@ def keyfunc(other):
       asm_setup += 'var setTempRet0 = Runtime.setTempRet0, getTempRet0 = Runtime.getTempRet0;\n'
 
     if settings['BINARYEN']:
-      table_size = sum(map(lambda table: table.count(',') + 1, last_forwarded_json['Functions']['tables'].values()))
-      asm_setup += "\nModule['wasmTableSize'] = %d;\n" % table_size
+      def table_size(table):
+        s = table.count(',')
+        return s + 1 if s > 0 else 0
```

Comment:
this doesn't seem right. If the table is `[just_this_one_function]` then we should return 1. I.e. `s` is set to 0 if the table size is either 0 or 1.

to detect an empty table, we should probably check for `table == '[]'`. I don't think you'll need to erase whitespace, we don't emit any